jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java

```diff
@@ -1953,4 +1987,56 @@ public X509 getX509()
             return _x509;
         }
     }
+
+    public static class Client extends SslContextFactory
+    {
+        public Client()
+        {
+            this(false);
```

Comment:
this constructor is deprecated.  So we need a new non deprecated protected constructor that can be used by both Client and Server.  So make the current `private SslContextFactory(boolean trustAll, String keyStorePath)`  a protected constructor and call it directly.
The old public constructors can then log warnings!

Reply:
The constructor is not deprecated. It is calling a new `Client(boolean)` constructor.